### PR TITLE
Slack notice

### PR DESCRIPTION
**Pull request by dubb-b**
### What does this PR do?
Adds slack notice to failed PR's in channel `#jenkins-prod-pr`

### What issues does this PR fix or reference?
New Feature

### Previous Behavior
None

### New Behavior
Awesome notifications now!!!

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by damon-atkins** [comment]
Try this out, but you will need to write the code I assume.  Given all tests need a working python you could use python to pull the info out of github.
https://api.github.com/repos/SaltStack/Salt/pulls/49831
Which will give you url for each reviewer "url": "https://api.github.com/users/gtmanfred",
Then in the message assume the github username and slack are the same and use @dubb-b in the slack message.

**Comment by dubb-b** [comment]
@damon-atkins Are you in community slack? If so hit me up, lets talk more about this. I am just trying to get a slack message for any PR that fails into a slack message so that we can start to identify them. But this is just a stop gap for now. We want to add some sort of TICK stack to Jenkins so that we get metrics on all builds, branch,  PR etc.. Then we can group the data better. But I would love to hear any ideas you have. I've noticed you comment about this before. This process is one of our top priorities. And any feedback is welcome.

In your suggestion for improving this do you think adding more data is relevant? I could have a script that pulls the reviewer into the message. And quite a bit more if needed. @rallytime would this help at all?

I am checking out for the eve, although Monday feel free to hit me up.

**Comment by rallytime** [comment]
> Then in the message assume the github username and slack are the same

We can't really assume that. There's lots of people who don't have the same GitHub and Slack usernames.

@dubb-b It seems here that you've got more commits in this than intended - this looks like you've got the stuff that has already been merged in here (i.e. the timeouts for docs/lint) rather than _just_ the slack additions. Maybe this needs a rebase? 

We can definitely chat more about this on Monday, but I think the general implementation is a great place to start.

File: .ci/docs [hunk 1]
```diff
@@ -45,6 +48,7 @@ pipeline {
                 description: 'The docs job has failed',
                 status: 'FAILURE',
                 context: "jenkins/pr/docs"
+            slackSend (channel: "#jenkins-prod-pr", color: '#FF0000', message: "FAILED: PR-Job: '${env.JOB_NAME} [${env.BUILD_NUMBER}]' (${env.BUILD_URL})" + "\n It's not denial. I'm just selective about the reality I accept.")
```
Comment (gtmanfred) [inline .ci/docs]: this indent looks wrong.

**Review by gtmanfred — CHANGES_REQUESTED** [review]
A few cosmetic changes.

**Review by rallytime — COMMENTED** [review]
Once this is rebased I can review this.

**Comment by rallytime** [comment]
@dubb-b Where are we with this one?

**Pull request closed without merging**